Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/SceneDynamicLoaderSystem.cs

```diff
@@ -0,0 +1,78 @@
+using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using ECS.Abstract;
+using ECS.Unity.Transforms.Components;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using Utility;
+
+namespace ECS.SceneLifeCycle.Systems
+{
+    [UpdateInGroup(typeof(SimulationSystemGroup))]
+    public partial class SceneDynamicLoaderSystem : BaseUnityLoopSystem
+    {
+        internal readonly SceneLifeCycleState state;
+
+        internal (UnityWebRequestAsyncOperation, List<Vector2Int>)? pointerRequest;
+
+        public SceneDynamicLoaderSystem(World world, SceneLifeCycleState state) : base(world)
+        {
+            this.state = state;
+        }
+
+        protected override void Update(float dt)
+        {
+            // TODO: Drop web request on realm change
+            if (!pointerRequest.HasValue)
+            {
+                // If we don't have a pointer request, we check the parcels in range, filter the parcels that are not loaded, and we create the request
+                var position = World.Get<TransformComponent>(state.PlayerEntity).Transform.position;
+
+                List<Vector2Int> parcelsToLoad = new List<Vector2Int>();
```

Comment:
The best way to avoid this transient memory alloc is `stackalloc` as you know the size of the array to allocate. Please refer to https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc.

But if you need to store it in a field just preallocate it once and reuse

Reply:
You know the maximum size, you can allocate an array big enough to hold data, count how many elements are really used, and then `Span.Slice` to use further

Reply:
Anyways, you have three ways of avoiding allocations:

- have a collection in a field, clear it, and reuse
- Use ListPool/HashSetPool
- allocate on stack

Keeping a temporary allocation is not an option